INSTALL.md

```diff
@@ -1,76 +1,221 @@
 
-If you are using `stack` or `nix` please make sure to add the latest version
-from Hackage to your tool configuration.
+**Note**: If you are new to Haskell you may find the [Haskell Getting
+Started
+Guide](https://github.com/composewell/haskell-dev/blob/master/getting-started.rst)
+to be useful.
 
-## Recommended build options
+If you know your way around Haskell, and have an up to date toolchain
+already installed, then you can jump straight to the section titled
+"[_Prepare Your Build Directory_](#user-content-prepare-your-build-directory)".
 
-See [recommended compilation options here](docs/Build.md).
+## Getting started with `streamly` using `cabal`
 
-## Using with `cabal`
+To get started, you will need a recent `cabal` binary to be installed
+on your system.
 
-Make sure you have `cabal` version 3.0 or later installed and you have `ghc`
-available in your PATH. Refresh your package list:
+### Installing `cabal`
+
+A pre-packaged `cabal` binary may already be available for your
+operating system.
+
+For example, on Debian GNU/Linux, you could install `cabal` using::
+
+```
+$ sudo apt install cabal-install
+```
+
+### Installing `ghc`
+
+You would also need to install the Glasgow Haskell Compiler `ghc`.  In
+most operating systems, using the system's package manager to install
+`cabal` would also bring in a version of `ghc` as a dependency.
+
+Please verify that `ghc` is present:
+
+```
+$ ghc --version
+The Glorious Glasgow Haskell Compilation System, version 8.0.2
+```
+
+If `ghc` isn't present, please install it using your operating
+system's package manager. For example, on Debian GNU/Linux, use:
+
+```
+$ sudo apt install ghc
+```
+
+### Update `cabal` if necessary
+
+Many operating systems ship an old version of `cabal`; in order to use
+`streamly` you should use `cabal` version 3.0 (or later).
+
+You can check the installed version of `cabal` by using its
+`--version` flag:
+
+```
+$ which cabal
+/usr/bin/cabal
+$ /usr/bin/cabal --version
+cabal-install version 1.24.0.2
+compiled using version 1.24.2.0 of the Cabal library
+```
+
+In this example, the installed `cabal` binary is too old for our
+needs. It would need to be upgraded.
+
+You can ask `cabal` to upgrade itself:
+
+```
+$ cabal install Cabal cabal-install
+Resolving dependencies...
+Downloading base16-bytestring-0.1.1.7...
+... lots of build output ...
+Building cabal-install-3.0.0.0...
+Installed cabal-install-3.0.0.0
+```
+
+`cabal` will usually place its compiled binaries inside the directory
+`$HOME/.cabal/bin`.
+
+You can verify the version of your newly installed binary by using the
+`--version` flag again:
+
+```
+$ $HOME/.cabal/bin/cabal --version
+cabal-install version 3.0.0.0
+compiled using version 3.0.2.0 of the Cabal library
+```
+
+If you have upgraded `cabal`, do be sure to specify `$HOME/.cabal/bin`
+early in your `PATH` shell variable, so that the new binary gets used
+instead of the older version on your system.
+
+```
+$ PATH=$HOME/.cabal/bin:$PATH
```

Comment:
We will also need to put that in `.bash_profile` or equivalent. `ghcup` takes care of that as well.